### PR TITLE
refactor with call & poll ema

### PR DESCRIPTION
**Pull request by aobo-y**
@jasonjabbour 

I noticed that you have copied & pasted the same set of codes to communicate with EMA in many places.
It becomes too redundant to follow. I have extracted them into two functions and applied them everywhere necessary
- `call_ema` send request to ema
- `poll_ema` poll the db for response

This should also help implement the retry timing logic
Please review & test

### CODE REVIEW

File: pkg/ema.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+def poll_ema(id, empathid, action_idx, duration=300, freq=5):
+  answer = None
+
+  try:
+    db = get_conn()
+    cursor = db.cursor()
+
+    var_name_code = str(action_idx + 1)
+    var_name_code = '0' * (3 - len(var_name_code)) + var_name_code
```
Comment (aobo-y) [inline pkg/ema.py]: @jasonjabbour  I don't quite sure about this `var_name_code `. does the logic work like a mapping about survey id
```
22 -> 000
19 -> 001
20 -> 002
21 -> 003
```

File: pkg/ema.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    send = urllib.request.urlopen(url)
+  except http.client.BadStatusLine: # EMA return non-TCP response
+    pass
+
+  # connect to database for logging
+  try:
+    db = get_conn()
+    cursor = db.cursor()
+
+    insert_query = "INSERT INTO reward_data(empathid,TimeSent,RecommSent,TimeReceived,Response,Uploaded) \
```
Comment (aobo-y) [inline pkg/ema.py]: always create a logging record about what we have sent
and later update it with response in the polling

File: pkg/ema.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+  except Exception as err:
+    log('Failed to log ema request:', err)
+    db.rollback()
+  finally:
+    db.close()
+
+  return empathid
+
+
+def poll_ema(id, empathid, action_idx, duration=300, freq=5):
```
Comment (aobo-y) [inline pkg/ema.py]: I believe doing `retry` may also need args like this, `duration` for how long and `freq` for how often

a naive retry idea is to write a new function using the `call_ema` and `poll_ema` here

```
while time < retry_duration:
  req_id = call_ema()
  ans = poll_ema(req_id, duration=retry_freq, freq=300)
  if ans:
    break
```

File: pkg/ema.py (hunk 1 above)
Comment (jasonjabbour) [reply pkg/ema.py]: Yes, we once discussed this in a meeting. The action id could be length 1,2, or 3. However, in the retrieval query we must always use three digits. "R_ _ _ Q01" 
Your solution works and is more concise. Thanks!

File: pkg/ema.py (hunk 2 above)
Comment (jasonjabbour) [reply pkg/ema.py]: got it!

**Pull request merged**